controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -345,74 +346,60 @@ private boolean deploysTo(Application application, JobType jobType) {
      * @param cause describes why the job is triggered
-     * @return the application in the triggered state, which *must* be stored by the caller
+     * @return the application in the triggered state, if actually triggered. This *must* be stored by the caller
      */
-    private Application triggerAllowParallel(JobType jobType, Application application, boolean first, String cause, Lock lock) {
-        if (jobType == null) { // previous was last job
-            return application;
-        }
-
-        // Note: We could make a more fine-grained and more correct determination about whether to block 
-        //       by instead basing the decision on what is currently deployed in the zone. However,
-        //       this leads to some additional corner cases, and the possibility of blocking an application
-        //       fix to a version upgrade, so not doing it now
-        if (jobType.isProduction() && application.deployingBlocked(clock.instant())) {
-            return application;
-        }
-        
-        if (application.deploymentJobs().isRunning(jobType, jobTimeoutLimit())) {
-            return application;
-        }
-
-        // TODO: Remove when we can determine why this occurs
-        if (jobType != JobType.component && ! application.deploying().isPresent()) {
-            log.warning(String.format("Want to trigger %s for %s with reason %s, but this application is not " +
-                                              "currently deploying a change",
-                                      jobType, application, cause));
-            return application;
-        }
-
-        if  ( ! deploysTo(application, jobType)) {
-            return application;
-        }
-
-        // Note that this allows a new change to catch up and prevent an older one from continuing
-        if ( ! application.deploymentJobs().isDeployableTo(jobType.environment(), application.deploying())) {
+    public Application triggerAllowParallel(JobType jobType, Application application, 
+                                            boolean first, boolean force, String cause, Lock lock) {
+        // Never allow untested changes to go through
+        // Note that this may happen because a new change catches up and prevents an older one from continuing
+        if ( jobType!= null && ! application.deploymentJobs().isDeployableTo(jobType.environment(), application.deploying())) {
             log.warning(String.format("Want to trigger %s for %s with reason %s, but change is untested", jobType,
                                       application, cause));
             return application;
         }
 
-        // Ignore applications that are not associated with a project
-        if ( ! application.deploymentJobs().projectId().isPresent()) {
-            return application;
-        }
-
+        if ( ! force && ! allowedTriggering(jobType, application)) return application;
```

Comment:
Move the 
```java
if (jobType == null) return ...
```
outside, lest you'll have NPEs below.

Reply:
That can not actually happen, but I agree it is clearer that way. Fixing ...